### DISCUSSION TITLE
There was a problem when I used JWT token state preservation。

### DISCUSSION THREAD

**Discussion by zizle** [fastapi/fastapi#9105] [Questions] [answered]
I'm using **version 0.58**
There may be an error in 'create'_ access_ Token 'function

There was a problem when I used JWT token state preservation。

In the example function of the official website, the user logs in to create a token：

```
def create_access_token(data: dict, expires_delta: Optional[timedelta] = None):
    to_encode = data.copy()
    if expires_delta:
        expire = datetime.utcnow() + expires_delta  # ------------------------------------------- this line
    else:
        expire = datetime.utcnow() + timedelta(minutes=15)  # ------------------------------------------- this line
    to_encode.update({"exp": expire})
    encoded_jwt = jwt.encode(to_encode, SECRET_KEY, algorithm=ALGORITHM)
    return encoded_jwt
```

In the validation function：

```
async def get_current_user(token: str = Depends(oauth2_scheme)):
    credentials_exception = HTTPException(
        status_code=status.HTTP_401_UNAUTHORIZED,
        detail="Could not validate credentials",
        headers={"WWW-Authenticate": "Bearer"},
    )
    try:
        payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORITHM])   # No matter how to verify, it can be verified correctly,
        username: str = payload.get("sub")
        if username is None:
            raise credentials_exception
        token_data = TokenData(username=username)
    except JWTError:
        raise credentials_exception
    user = get_user(fake_users_db, username=token_data.username)
    if user is None:
        raise credentials_exception
    return user
```

No matter how to verify, it can be verified correctly, and the set failure time has no effect. But when I change the expiration time to：

>  **expire = time.time() + 900  # 900 is seconds**

it works!

I don't know if it's a problem, if it's OK. Just ignore it.

**Comment by tiangolo** [comment]
@zizle were you having problems with it? Was it not working as expected?

**Comment by zizle** [comment]
> @zizle were you having problems with it? Was it not working as expected?

In login logic, create a user token: expire = datetime.utcnow() + timedelta(minutes=15) It never expired.
but,this way: expire = time.time() + 900 # seconds=900 It correctly。

I don't know if it's a problem, if it's OK. Just ignore it.

**Comment by YuriiMotov** [comment, marked as answer]
Code example in docs works correctly.
Try modifying `ACCESS_TOKEN_EXPIRE_MINUTES` constant to `1`, restart the server. Then go to Swagger interface, authorize using green button, wait for 1-2 minutes (at least 1 minute) and try accessing any of the protected endpoints. You will get `Could not validate credentials` response with the status code `401`
